xmake/core/tool/compiler.lua

```diff
@@ -335,5 +335,27 @@ function compiler:compflags(opt)
     return self:_preprocess_flags(flags)
 end
 
+-- filter the compiling flags by removing those who don't should be part of depends files
+--
+-- @param flags   flags list
+--
+-- @return      a new filtered flags list
+--
+function compiler:depflags(flags)
+    local nondepflags = self:get("nondepflags")
+    if nondepflags == nil then
+        return table.copy(flags)
+    end
+
+    local filteredflags = {}
+    for _, flag in ipairs(flags) do
+        if not nondepflags[flag] then
+            table.insert(filteredflags, flag)
+        end
+    end
+
+    return filteredflags
```

Comment:
This will affect compilation performance, because every time we compile a source file, we need to traverse all the flags additionally.

So I haven't tried this before.